editor/animation_editor.cpp

```diff
@@ -2995,6 +3000,56 @@ void AnimationKeyEditor::_notification(int p_what) {
 			_update_menu();
 
 		} break;
+
+		case EditorSettings::NOTIFICATION_EDITOR_SETTINGS_CHANGED: {
+
+			zoomicon->set_texture(get_icon("Zoom", "EditorIcons"));
```

Comment:
is this code, all repeated somewhere else, wouldnt it be better to create a function.

Reply:
Yea I was discussion that another day. if we might should introduce functions for the icon and style box changes. since they usually are needed in enter_tree and settings_chagned.

**maybe it is possible to listen to theme_changed**. since that is also a control event and should be called on tree enter and when the settings changes the tree!

Reply:
I'm just doing that right now, that I move all the icon updates in theme changed. this just makes sooo much more sense than editor settings changed. editor settings sometimes change the theme and if they do the theme updates that is just soo much better!

I might do that for all the editor classes soon. since that also saves performance for normal settings changes. and makes sure the theme always updates. also than it is not needed to have it in settings changed + enter tree... as mentioned before so we also can get rid of a lot or redundant code...

Reply:
@toger5 yeah, that sounds really good, however i would suggest you to wait until most breaking PRs get merged, so you have less conflicts